### ISSUE TITLE
Document retrieved is different from one time to another

### ISSUE THREAD

**Issue by clementbiron** [ambanum/OpenTermsArchive#805]
The document retrieved for SHEIN Commercial Terms is sometimes the mobile version, sometimes the desktop version. View the associated diff https://github.com/OpenTermsArchive/versions-france/commit/e4742be0c637500454bf2f8c0d1b522fad7dff00 and https://github.com/OpenTermsArchive/versions-france/commit/a9cca60e67e855c66009b823b6b5f1c94793ff6d

As changes are discovered, we need to ensure that we consistently provide the same version of a document.

**Comment by clementbiron** [comment]
Yes 👍

**Comment by MattiSG** [comment]
Indeed, it would be good to restrict the device category to stabilise the documents we receive 🙂
But, more generally, why do we generate a random user agent altogether? I understand the intention, but do we have any idea how many documents less we would get by using a legit `Open Terms Archive` user agent? 🙂

**Comment by MattiSG** [comment]
I wonder how much this is also the source of the blinking Meta pages such as in https://github.com/OpenTermsArchive/france-elections-versions/commit/709edec29cd516aa33bee475a7bea071d9fae029. Indeed, beyond the `?locale2=fr_FR` query strings, I also see that some [destination domains](https://github.com/OpenTermsArchive/france-elections-versions/commit/709edec29cd516aa33bee475a7bea071d9fae029#diff-e43a59590f4f051a818725f7b8453d6db4b26ce8ad524d71bc8d8293d72bc28cR248) switch to a `m` subdomain.

**Comment by Ndpnt** [comment]
I am in favour of removing the random user agent generation. Also, I suggest avoiding randomness in code as much as possible: Random things lead to random bugs, which are very difficult to debug.

**Comment by martinratinaud** [comment]
This is part of best practices of scraping

https://www.blog.datahut.co/post/web-scraping-best-practices-tips

Also, I think I implemented it because I had problems on a declaration (but did not document it)

i'd rather fix it to one version than remove it completely, or use "OpenTermsArchive/v1" but I'm pretty sure many servers will kick us out with this

**Comment by clementbiron** [comment]
By removing the user agent generation, the risk of getting kicked increases considerably.

**Comment by Ndpnt** [comment]
Currently, random generation of the user agent is only performed for documents retrieved with the headless browser. If we remove it, it can only affect these documents, which is :
- 72/607 (~12%) documents for `contrib`
- 38/209 (~18%) documents for `france`.
- 6/60 (10%) documents for `dating 
- 4/65 (~6%) documents for `france-elections`.

Thus, the overall impact can only affect ~13% of documents.

This also means that, at the moment, we don't have random user agent for 87% of our documents and we don't get kicked that much.

I also did some testing on the experimental server with the `contrib` declarations where I removed the random generation of the user agent and compared whether there were more inaccessible documents than currently on the production server, and the result is that, apart from the minor variations we may have between each run, there is absolutely no difference.

**Comment by martinratinaud** [comment]
Ok, then I agree to remove it and we'll see if we need to put it back later.
